packages/typescript-to-proptypes/src/getPropTypesFromFile.ts

```diff
@@ -114,7 +114,10 @@ function checkType({
         return createInstanceOfType({ jsDoc, instance: 'RegExp' });
       }
       case 'Date': {
-        return createInstanceOfType({ jsDoc, instance: 'Date' });
+        if (!project.shouldUseObjectForDate?.({ name })) {
+          return createInstanceOfType({ jsDoc, instance: 'Date' });
+        }
+        break;
```

Comment:
To avoid relying on the next part of the script that could be modified one day
```suggestion
        }
        return createObjectType({ jsDoc });
```